lib/mcb/courses_editor.rb

```diff
@@ -59,22 +78,63 @@ def ask_gcse_subject(subject)
       end
     end
 
-    def edit_route
-      print_existing(:program_type)
-      update(program_type: ask_route)
-    end
-
     def ask_route
       @cli.choose do |menu|
         menu.prompt = "What's the route?  "
         menu.choices(*Course.program_types.keys)
       end
     end
 
+    def ask_qualifications
+      @cli.choose do |menu|
+        menu.prompt = "What's the course outcome?  "
+        menu.choices(*Course.qualifications.keys)
+        menu.default = "pgce_with_qts"
+      end
+    end
+
+    def ask_study_mode
+      @cli.choose do |menu|
+        menu.prompt = "Full time or part time?  "
+        menu.choices(*Course.study_modes.keys)
+        menu.default = "full_time"
+      end
+    end
+
+    def ask_accredited_body
+      new_accredited_body = nil
+      until new_accredited_body.present?
+        begin
+          new_accredited_body = ask_accredited_body_once
+        rescue ActiveRecord::RecordNotFound
+          puts "Can't find accredited body; please enter one that exists."
+        end
+      end
+      new_accredited_body
+    end
+
+    def ask_accredited_body_once
+      code = @cli.ask "Provider code of accredited body (leave blank if self-accredited)  ", ->(str) { str.upcase }
+      code.present? ? Provider.find_by!(provider_code: code) : @provider
+    end
+
+    def ask_start_date
+      Date.parse(@cli.ask("Start date?  ") { |q| q.default = "September #{@courses.first.recruitment_cycle}" })
+    end
+
+    def ask_application_opening_date
```

Comment:
is this related to 

https://github.com/DFE-Digital/manage-courses-backend/blob/2794cc115207c201a45b06fccda2d138689bbe67/app/models/course.rb#L150

if so can we rename the method 'ask_applications_open_from'

Reply:
> if so can we rename the method 'ask_applications_open_from'

The different names is a deliberate choice – the `ask_*` method names refer to the domain logic naming and the language we're using in the frontend, and these map to the database names using [this lovely mapping](https://github.com/DFE-Digital/manage-courses-backend/blob/ca0d01a3cb49fec9c6c867edd6cf35b42c5f4dd9/lib/mcb/courses_editor.rb#L3-L14) that I'm introducing as part of this PR.

I'm hoping that we eventually can rename the database fields to be in line with the language that we're using in the frontend, so I don't want the database names to be embedded everywhere, but that's a personal preference.

> 
> btw how does that one date get associated to a single site?

This PR defines a setter on `Course` here: https://github.com/DFE-Digital/manage-courses-backend/blob/ca0d01a3cb49fec9c6c867edd6cf35b42c5f4dd9/app/models/course.rb#L163-L165

Reply:
I'm gonna merge this PR so we can keep dismantling the extra tooling branch. Happy to discuss naming in person tomorrow and can make naming consistent with the database field names if that's the consensus opinion. Ultimately it shouldn't matter in the long run because:

1. this code will reduce and eventually disappear completely in future, as more and more functionality is moved to be self-serve
2. a lot of the `site_status` malarkey will also disappear as we simplify the data model.